### ISSUE TITLE
Feature Request - IP Sort Order

### ISSUE THREAD

**Issue by cspacone** [angryip/ipscan#164]
under Tools/Preferences/Display add sticky preference - 

Sort Order: Natural [ ], Ascending [ ], Descending [ ]

Where 
Natural [X] - sets the current default sort order behaviour of AngryIP
Ascending [X] -  sets sort order from lowest IP to highest IP
Descending [X] - sets sort order from highest IP to lowest IP

Apologies if this has already been asked for and deferred or rejected.

Excellent work!

**Comment by angryziber** [comment]
Hi! You can already sort like this by pressing on the IP header, but only the scan has finished.
Reordering is quite hard to achieve while scan is running, so this feature is disabled at that time.

**Comment by cspacone** [comment]
Apply the setting after the sort? I don't expect the scanner to sort
on the fly as it is scanning, just change the default display once the
scan / threads complete.
But as you say, I could always do a pull and code it myself, except
that I'm lacking in those skills.
Anton, thank you for your scanner, it has saved me loads of time and
is very convenient.
-Chris

	-----------------------------------------From: "Anton Keks" 
To: "angryip/ipscan"
Cc: "cspacone", "Author"
Sent: Monday December 24 2018 4:30:29AM
Subject: Re: [angryip/ipscan] Feature Request - IP Sort Order (#164)

	Hi! You can already sort like this by pressing on the IP header, but
only the scan has finished.
 Reordering is quite hard to achieve while scan is running, so this
feature is disabled at that time. 

	—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub [1], or mute the
thread [2]. 

Links:
------
[1]
https://github.com/angryip/ipscan/issues/164#issuecomment-449728531
[2]

**Comment by angryziber** [comment]
Does the click on IP header work for you?

**Comment by cspacone** [comment]
Yes, it works fine. What I had hoped to make the case for (and clearly failed to do so) was setting the post-scan behavior of the IP list to sort (natural, ascending or descending) based on a setting in preferences.

 

Please feel free to ignore and close this request.

 

-Chris

**Comment by marxenegls** [comment]
I have the same issue, i just can't go through all the ip's listed in the settings tab , the cursor doesn't move any further nor those the window enlarge .I can't se which ports are listed by default nor can i edit them

**Comment by anasazi-970** [comment]
Ever since using the earliest Windows versions that did sort at the end (it seemed?) I have missed
this feature.  I'm a Linux user now days.  A post sort option as suggested in this thread would be
superb to an already great utility.  I am so used to scanning (with my eyes) down the list to see if
all my little devices and kiosk's and NAS's are up and running by saying hi to them.
I looked at the source code, I probably could do it, I saw a call to a sort function in the ping fetcher
code file.  What I couldn't find is a place in the main line code that would know the sort is done
and puts up that 'stats box' at the end.  Would love some pointers on doing it...
Thx. All

**Comment by JoeBarouneD** [comment]
I agree, manually clicking to sort after a scan is annoying ...

**Comment by OskarZyg** [comment]
I think the proper way to fix this would be to remember the user preference on how the list was sorted last time, and by what. I believe the reason this doesn't happen on windows is because the OS does that there, but this behaviour doesn't exist on Linux and macOS.
